### PR TITLE
REFS:rx-training#44 and fixes rx-training#44 (commit day15)

### PR DESCRIPTION
**Pull request by karansinghchauhan63**

### CODE REVIEW

File: Modules/JavaScript/Day15/Karan-Chauhan/assignment/assignment.html [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            <div class="form-group">
+                <input type="button" onclick="myOperation()" value="Calculate" id="calculate" class="btn btn-primary">
+                <input type="reset" value="Reset" id="reset" class="btn btn-secondary">
+            </div>
+            <label id="result"></label>
+        </form>
+    </div>
+</body>
+
+</html>
```
Comment (MilitPanchasara) [inline Modules/JavaScript/Day15/Karan-Chauhan/assignment/assignment.html:85]: Also reset the result and adding validations on input can prevent NaN result value.

File: Modules/JavaScript/Day16/Karan-Chauhan/assignment/assignment.html [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            <form id="form1">
+                <h3 class="display-5">CHECK ENTERED DATE IS VALID OR NOT:</h3>
+                <label for="string">Enter the Date in "MM-DD-YYYY" format:</label>
+                <input type="text" class="form-control" id="date" pattern="[0-9]{10}" required>
+                <input type="submit" value="CHECK" class="btn btn-warning m-2" onclick="return checkDate()">
+            </form>
+        </div>
+    </div>
+    <br><br><hr>
+</html>
```
Comment (MilitPanchasara) [inline Modules/JavaScript/Day16/Karan-Chauhan/assignment/assignment.html:66]: No leap year validations, validation for February is not there, even 02-30-2021 is valid date according to the code.

File: Modules/JavaScript/Day18/Karan-Chauhan/assignment/assignment.html [hunk 3]
```diff
@@ -11,122 +11,28 @@
...
-                <p id="p7"></p> 
-                <p id="p8"></p>                   
-            </div>
-        </div>
+        <div class="alert alert-danger" id="demo"></div>        
     </div>
-    <br><br>
-    <hr>
-
 </html>
```
Comment (MilitPanchasara) [inline Modules/JavaScript/Day18/Karan-Chauhan/assignment/assignment.html:38]: Promises/Callback is not used

File: Modules/JavaScript/Day20/Karan-Chauhan/assignment/assignment.html [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+            </table>
+            <br>
+            <button class="btn btn-warning" id="cartSum" onclick="cartSummary()">Cart Summary</button>
+        </div>
+
+        
+    </div>
+    </body>
+
+</html>
```
Comment (MilitPanchasara) [inline Modules/JavaScript/Day20/Karan-Chauhan/assignment/assignment.html:156]: Multiple items are not getting stored in cart, If I add one, previous one is getting replaced by new one.
You can add a item object to a array and then store that whole array in localStorage.

File: Modules/JavaScript/Day16/Karan-Chauhan/assignment/assignment.html (hunk 2 above)
Comment (karansinghchauhan63) [reply Modules/JavaScript/Day16/Karan-Chauhan/assignment/assignment.html:66]: Query resolved

File: Modules/JavaScript/Day15/Karan-Chauhan/assignment/assignment.html (hunk 1 above)
Comment (karansinghchauhan63) [reply Modules/JavaScript/Day15/Karan-Chauhan/assignment/assignment.html:85]: Query Resolved

File: Modules/JavaScript/Day18/Karan-Chauhan/assignment/assignment.html (hunk 3 above)
Comment (karansinghchauhan63) [reply Modules/JavaScript/Day18/Karan-Chauhan/assignment/assignment.html:38]: Query Resolved

File: Modules/JavaScript/Day20/Karan-Chauhan/assignment/assignment.html (hunk 4 above)
Comment (karansinghchauhan63) [reply Modules/JavaScript/Day20/Karan-Chauhan/assignment/assignment.html:156]: All Queries Resolved (Final Commit)

**Pull request merged**